### PR TITLE
[Android] Fixed receiving of referrer from GPS

### PR DESCRIPTION
**Pull request by samartnik**
Resolves https://github.com/brave/brave-browser/issues/15903

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: patches/chrome-android-java-AndroidManifest.xml.patch [hunk 1]
```diff
@@ -45,7 +45,7 @@ index e5a9fd2d98b65ff7bb54ded343d1de3891b232a8..70a15ac085e9c2dd413d73b13d252dfe
  
 +<!--
      <uses-permission android:name="com.google.android.apps.now.CURRENT_ACCOUNT_ACCESS" />
-+-->
++--> <uses-permission android:name="com.google.android.finsky.permission.BIND_GET_INSTALL_REFERRER_SERVICE"/>
```
Comment (samartnik) [inline patches/chrome-android-java-AndroidManifest.xml.patch:48]: This is for a quick fix as we need it ASAP, follow up issue for proper AndroidManifest patching is created https://github.com/brave/brave-browser/issues/15942

**Review by mkarolin — APPROVED** [review]
`patches` ++

**Review by keur — APPROVED** [review]

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by jumde — APPROVED** [review]
lgtm

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Samsung S10+` running `Android 11` using the following build:

```
1.26.31 CR: 91.0.4472.57
```

STR:

* visited https://kjozwiak.github.io/
* clicked on `Nightly GPS TVT141` which points to the following:
  * `https://play.google.com/store/apps/details?id=com.brave.browser_nightly&referrer=urpc%3DTVT141`
 * installed Nightly via GPS
 * launched and checked `brave://local-storage`
 
 ```
 "referral": {
         "download_id": "69816d5c-cd0c-496c-9911-a42926558462",
         "headers": [ {
            "cookieNames": [  ],
            "domains": [ "eaff.com", "stg.eaff.com" ],
            "expiration": 31536000000.0,
            "headers": {
               "X-Brave-Partner": "eaff"
            }
         }, {
            "cookieNames": [  ],
            "domains": [ "sandbox.uphold.com", "api-sandbox.uphold.com", "uphold.com", "api.uphold.com" ],
            "expiration": 31536000000.0,
            "headers": {
               "X-Brave-Partner": "uphold"
            }
         }, {
            "cookieNames": [  ],
            "domains": [ "www.grammarly.com", "grammarly.com", "static.grammarly.com", "gnar.grammarly.com" ],
            "expiration": 31536000000.0,
            "headers": {
               "X-Brave-Partner": "grammarly"
            }
         } ],
         "initialization": true,
         "promo_code": "TVT141"
      },
      "referral_android_first_run_timestamp": "13265929012491584",
      "stats": {
         "first_check_made": true,
         "last_check_month": 5,
         "last_check_woy": 20,
         "last_check_ymd": "2021-05-19",
         "threshold_query": "https://laptop-updates.brave.com/1/usage/brave-core-threshold?platform=android-bc&channel=canary&version=1.26.31&daily=true&weekly=true&monthly=true&first=true&woi=2021-05-17&dtoi=2021-05-19&ref=TVT141&adsEnabled=false&arch=",
         "week_of_installation": "2021-05-17"
      }
```

Screenshot_20210519-161723_Brave - Nightly

**Pull request merged**